filebeat/input/log/harvester.go

```diff
@@ -644,10 +646,17 @@ func (h *Harvester) newLogFileReader() (reader.Reader, error) {
 		return nil, err
 	}
 
+	// Configure MaxBytes limit for EncodeReader as multiplied by 4
+	// for the worst case scenario where incoming UTF32 charchers are decoded to the single byte UTF-8 characters.
+	// This limit serves primarily to avoid memory bload or potential OOM with expectedly long lines in the file.
+	// The further size limiting is performed by LimitReader at the end of the readers pipeline as needed.
+	encReaderMaxBytes := h.config.MaxBytes * 4
```

Comment:
may want to add a comment in docs for max_bytes that it can use up to 4x the number worth of memory for buffering

Reply:
Sure can do that. Which docs? Before this change it was taking all available memory :-)

Reply:
https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-log.html

As a user I would want that information here


The maximum number of bytes that a single log message can have. All bytes after max_bytes are discarded and not sent., _but up to 4x max_bytes will be buffered in-memory_ .This setting is especially useful for multiline log messages, which can get large. The default is 10MB (10485760).

Reply:
On the second thought I'm not sure that this part of internal implementation should be documented. 
Before this change the code was allocating as much as it was available and cutting the final message to the max byte limit before the output from the pipeline.
This change adds a protection from unlimited allocation.

I'll let the regular beats contributors that are going to review this PR to decide if this needs to be documented.

Reply:
Fair enough. I will make a quick pitch and leave it to you and reviewers to decide: without that kind of information, it is hard to plan. For example, how can I determine how much memory I will be safe giving filebeat? I may think that if I tell it to read only 1 file at a time, and tell it to use max_bytes of 25MB, that I am very safe giving it 50MB of memory, only to be sad when it OOMs because it actually used 100MB of memory for 1 line that I thought was limited to 25.

Personally I find it already very hard to figure out how much memory it needs because i have to do match like <overhead> + max_events * num_harvesters * max_size - and im scared of more hidden values (like * 4)